extensions/INTEL/SPV_INTEL_fpga_argument_interfaces.asciidoc

```diff
@@ -0,0 +1,184 @@
+2+| Literal Number (32-bit signed integer) +
+_AddressWidth_ | *FPGAArgumentInterfacesINTEL*
+// --- ROW BREAK ---
+| 6178 | *MMHostInterfaceDataWidthINTEL* +
+Must be applied only to a pointer *OpFunctionParameter* of a function that is an *entry point*. Indicates the size, in bits, of the data bus for the Memory Mapped Interface created for this pointer argument.
+2+| Literal Number (32-bit signed integer) +
+_DataWidth_ | *FPGAArgumentInterfacesINTEL*
+// --- ROW BREAK ---
+| 6179 | *MMHostInterfaceLatencyINTEL* +
+Must be applied only to a pointer *OpFunctionParameter* of a function that is an *entry point*. Indicates the latency in cycles of the Memory Mapped Interface created for this pointer argument. If this decoration is present it guarantees that the latency is fixed.
+2+| Literal Number (32-bit signed integer) +
+_Latency_ | *FPGAArgumentInterfacesINTEL*
+// --- ROW BREAK ---
+| 6180 | *MMHostInterfaceReadWriteModeINTEL* +
+Must be applied only to a pointer *OpFunctionParameter* of a function that is an *entry point*. Indicates the read-write mode of the Memory Mapped Interface created for this pointer argument.
+2+| *Access Qualifier* +
+_ReadWriteMode_ | *FPGAArgumentInterfacesINTEL*
+// --- ROW BREAK ---
+| 6181 | *MMHostInterfaceMaxBurstINTEL* +
+Must be applied only to a pointer *OpFunctionParameter* of a function that is an *entry point*. Indicates the maximum burst count of the Memory Mapped Interface created for this pointer argument.
+2+| Literal Number (32-bit signed integer) +
+_MaxBurstCount_ | *FPGAArgumentInterfacesINTEL*
+// --- ROW BREAK ---
+| 6182 | *MMHostInterfaceWaitRequestINTEL* +
+Must be applied only to a pointer *OpFunctionParameter* of a function that is an *entry point*. Indicates whether the Memory Mapped Interface created for this pointer argument should accept a waitrequest signal.
+
+A setting of 1 means build a waitrequest signal and a setting of 0 means don't.
+
+2+| Literal Number (32-bit signed integer) +
+_Waitrequest_ | *FPGAArgumentInterfacesINTEL*
+// --- ROW BREAK ---
+| 6183 | *StableKernelArgumentINTEL* +
+Must be applied only to an *OpFunctionParameter* of a function that is an *entry point*. Indicates that this input will not change during the execution of piplined kernel invocations. Input can change once all active invocations have finished.
+2+| | *FPGAArgumentInterfacesINTEL*
+|====
+--
+
+=== Capability
+
+Modify Section 3.31, Capability, adding a row to the Capability table:
+--
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 6174
+| FPGAArgumentInterfacesINTEL
+|
+|====
+--
+
+=== Validation Rules
+
+It is invalid to specify multiple copies of the same decoration on the same *OpFunctionParameter*. For example, you can't specify two *MMHostInterfaceAddressWidthINTEL* decorations on the same *OpFunctionParameter* with conflicting values.
```

Comment:
Mentioned in last week's SPIR-V call: this condition is already described generally in the main SPIR-V specification so it does not need to be duplicated here.  From [3.20 Decoration](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_decoration):

> It is invalid to decorate any given id or structure member more than one time with the same decoration, unless explicitly allowed below for a specific decoration.

It would be best to remove this validation rule to eliminate the duplication.

It's fine to keep the following validation rule ("it is invalid to specify both ConduitKernelArgumentINTEL and RegisterMapKernelArgumentINTEL decorations on the same OpFunctionParameter") because it isn't already covered by the main SPIR-V specification.

Thanks!

Reply:
done